sdk/monitor/monitor-opentelemetry-exporter/src/export/statsbeat/longIntervalStatsbeatMetrics.ts

```diff
@@ -0,0 +1,286 @@
+      longIntervalMetricReaderOptions
+    );
+    this._longIntervalStatsbeatMeterProvider.addMetricReader(this._longIntervalMetricReader);
+    this._longIntervalStatsbeatMeter = this._longIntervalStatsbeatMeterProvider.getMeter(
+      "Azure Monitor Long Interval Statsbeat"
+    );
+
+    // Assign Common Properties
+    this._runtimeVersion = process.version;
+    this._language = STATSBEAT_LANGUAGE;
+    this._version = ai.packageVersion;
+    this._cikey = options.instrumentationKey;
+
+    this._featureStatsbeatGauge = this._longIntervalStatsbeatMeter.createObservableGauge(
+      StatsbeatCounter.FEATURE
+    );
+    this._attachStatsbeatGauge = this._longIntervalStatsbeatMeter.createObservableGauge(
+      StatsbeatCounter.ATTACH
+    );
+
+    this._commonProperties = {
+      os: this._os,
+      rp: this._resourceProvider,
+      cikey: this._cikey,
+      runtimeVersion: this._runtimeVersion,
+      language: this._language,
+      version: this._version,
+      attach: this._attach,
+    };
+
+    this._attachProperties = {
+      rpId: this._resourceIdentifier,
+    };
+
+    this._isInitialized = true;
+    this._initialize();
+  }
+
+  private async _initialize() {
+    try {
+      await this._getResourceProvider();
+
+      // Add long interval observable callbacks
+      this._attachStatsbeatGauge.addCallback(this._attachCallback.bind(this));
+      this._longIntervalStatsbeatMeter.addBatchObservableCallback(
+        this._featureCallback.bind(this),
+        [this._featureStatsbeatGauge]
+      );
+
+      // Export Feature/Attach Statsbeat once upon app initialization
+      this._longIntervalAzureExporter.export(
+        (await this._longIntervalMetricReader.collect()).resourceMetrics,
+        (result) => {
+          if (result.code !== ExportResultCode.SUCCESS) {
+            diag.error(`LongIntervalStatsbeat: metrics export failed (error ${result.error})`);
+          }
+        }
+      );
+    } catch (error) {
+      diag.debug("Call to get the resource provider failed.");
+    }
+  }
+
+  private _featureCallback(observableResult: BatchObservableResult) {
+    let attributes;
+    if (this._instrumentation) {
+      attributes = {
+        ...this._commonProperties,
+        feature: this._instrumentation,
+        type: StatsbeatFeatureType.INSTRUMENTATION,
+      };
+      observableResult.observe(this._featureStatsbeatGauge, 1, { ...attributes });
+    }
+
+    if (this._feature) {
+      attributes = {
+        ...this._commonProperties,
+        feature: this._feature,
+        type: StatsbeatFeatureType.FEATURE,
+      };
+      observableResult.observe(this._featureStatsbeatGauge, 1, { ...attributes });
+    }
+  }
+
+  private _attachCallback(observableResult: ObservableResult) {
+    let attributes = { ...this._commonProperties, ...this._attachProperties };
+    observableResult.observe(1, attributes);
+  }
+
+  public async getAzureComputeMetadata(): Promise<boolean> {
```

Comment:
Please do not duplicate code, if you have shared code like this one to get resource provide details and other ones related to shortening endpoints or similar have them in a single shared place

Reply:
Modified this such that the `longIntervalStatsbeatMetrics` and `networkStatsbeatMetrics` extend `StatsbeatMetrics` which provides access to shared methods like these.